cmd/helm/completion.go

```diff
@@ -74,173 +74,23 @@ func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
 	if !found {
 		return errors.Errorf("unsupported shell type %q", args[0])
 	}
+	// In case the user renamed the helm binary (e.g., to be able to run
+	// both helm2 and helm3), we overwrite the new binary name to the command
+	// in order to generate proper completion
+	rootCmd := cmd.Root()
+	rootCmd.Use = filepath.Base(os.Args[0])
```

Comment:
I'm a little concerned about unknown side-effects of doing this.
But it may be ok.  I'm just leaving a note to keep thinking about it.

Reply:
We might want to create a deep copy of the command?
I haven't investigated much as if the command was still used afterwards.

Reply:
Simpler than a deep-copy could be to
- set Use field
- generate completion script
- reset Use field to old value.

But all this feels hacky to me.

Reply:
Seems hacky to me too, but i think that's a job for cobra, it definitly should be configurable.
Meanwhile we might want something like that.

I changed the way it was done in order to save/restore the Use field in fe31b1db37f1f1ee7452234a852aa286e2aea581